core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+    """
+    blog_post_rights_models = (
+        blog_models.BlogPostRightsModel.get_by_user(user_id))
+    return [
+        get_blog_post_rights_from_model(model)
+        for model in blog_post_rights_models
+        if model is not None]
+
+
+def does_blog_post_with_url_fragment_exist(url_fragment):
+    """Checks if blog post with provided url fragment exists.
+
+    Args:
+        url_fragment: str. The url fragment for the blog post.
+
+    Returns:
+        bool. Whether the the url fragment for the blog post exists.
+
+    Raises:
+        Exception. Blog Post URL fragment is not a string.
+    """
+    if not isinstance(url_fragment, python_utils.BASESTRING):
+        raise utils.ValidationError(
+            'Blog Post URL fragment should be a string. Recieved:'
+            '%s' % url_fragment)
+    existing_blog_post = get_blog_post_by_url_fragment(url_fragment)
+    return existing_blog_post is not None
+
+
+def _save_blog_post(blog_post):
+    """Saves a BlogPost domain object to the datastore.
+
+    Args:
+        blog_post: BlogPost. The blog post domain object for the given
+            blog post.
+    """
+    model = blog_models.BlogPostModel.get(blog_post.id, strict=True)
+    blog_post.validate()
+
+    model.title = blog_post.title
+    model.content = blog_post.content
+    model.tags = blog_post.tags
+    model.published_on = blog_post.published_on
+    model.thumbnail_filename = blog_post.thumbnail_filename
+    model.url_fragment = blog_post.url_fragment
+    model.update_timestamps()
+    model.put()
+
+
+def publish_blog_post(blog_post_id):
+    """Marks the given blog post as published.
+
+    Args:
+        blog_post_id: str. The ID of the given blog post.
+
+    Raises:
+        Exception. The given blog post does not exist.
+    """
+    blog_post_rights = get_blog_post_rights(blog_post_id, strict=False)
+    if blog_post_rights is None:
+        raise Exception('The given blog post does not exist')
+    blog_post = get_blog_post_by_id(blog_post_id, strict=True)
+    blog_post.validate(strict=True)
+    blog_post_summary = get_blog_post_summary_by_id(blog_post_id, strict=True)
+    blog_post_summary.validate(strict=True)
+    blog_post_rights.blog_post_is_published = True
+
+    published_on = datetime.datetime.utcnow()
+    blog_post.published_on = published_on
+    blog_post_summary.published_on = published_on
+
+    save_blog_post_rights(blog_post_rights)
+    _save_blog_post_summary(blog_post_summary)
+    _save_blog_post(blog_post)
+
+
+def unpublish_blog_post(blog_post_id):
+    """Marks the given blog post as unpublished or draft.
+
+    Args:
+        blog_post_id: str. The ID of the given blog post.
+
+    Raises:
+        Exception. The given blog post does not exist.
+    """
+    blog_post_rights = get_blog_post_rights(blog_post_id, strict=False)
+    if blog_post_rights is None:
+        raise Exception('The given blog post does not exist')
+    blog_post_rights.blog_post_is_published = False
```

Comment:
Do we need to update the summary and blog model as well?

Reply:
No we dont. Published_on field will be now be shown as 'last published on'. So no changes in summary or blog post model is required.